src/app/zap-templates/zcl/data-model/chip/joint-fabric-datastore-cluster.xml

```diff
@@ -0,0 +1,370 @@
+    <cluster code="0x0752"/>
+    <item fieldId="0" name="State" type="DatastoreStateEnum" default="0x00" min="0x00" max="0x03"/>
+    <item fieldId="1" name="UpdateTimestamp" type="epoch_s"/>
+    <item fieldId="2" name="FailureCode" type="status" default="0"/>
+  </struct>
+
+  <struct name="DatastoreNodeKeySetEntry" apiMaturity="provisional" isFabricScoped="true">
+    <cluster code="0x0752"/>
+    <item fieldId="0" name="NodeID" type="node_id"/>
+    <item fieldId="1" name="GroupKeySetID" type="int16u"/>
+    <item fieldId="2" name="StatusEntry" type="DatastoreStatusEntryStruct"/>
+  </struct>
+
+  <enum name="DatastoreAccessControlEntryPrivilegeEnum" type="enum8">
+    <cluster code="0x0752"/>
+    <item name="View" value="0x01"/>
+    <item name="ProxyView" value="0x02"/>
+    <item name="Operate" value="0x03"/>
+    <item name="Manage" value="0x04"/>
+    <item name="Administer" value="0x05"/>
+  </enum>
+
+  <struct name="DatastoreGroupInformationEntryStruct" isFabricScoped="true">
+    <cluster code="0x0752"/>
+    <item fieldId="0" name="GroupID" type="int64u"/>
+    <item fieldId="1" name="FriendlyName" type="char_string" length="32"/>
+    <item fieldId="2" name="GroupKeySetID" type="int16u" isNullable="true" min="1" max="65534"/>
+    <item fieldId="3" name="GroupCAT" type="int16u" isNullable="true" min="1" max="65534"/>
+    <item fieldId="4" name="GroupCATVersion" type="int16u" isNullable="true" min="1" max="65534"/>
+    <item fieldId="5" name="GroupPermission" type="DatastoreAccessControlEntryPrivilegeEnum" min="0x00" max="0x05"/>
+  </struct>
+
+  <struct name="DatastoreBindingTargetStruct">
+    <cluster code="0x0752"/>
+    <item fieldId="1" name="Node" type="node_id" optional="true"/>
+    <item fieldId="2" name="Group" type="group_id" optional="true" min="1"/>
+    <item fieldId="3" name="Endpoint" type="endpoint_no" optional="true"/>
+    <item fieldId="4" name="Cluster" type="cluster_id" optional="true"/>
+  </struct>
+
+  <struct name="DatastoreEndpointBindingEntryStruct" isFabricScoped="true">
+    <cluster code="0x0752"/>
+    <item fieldId="0" name="NodeID" type="node_id"/>
+    <item fieldId="1" name="EndpointID" type="endpoint_no"/>
+    <item fieldId="2" name="ListID" type="int16u"/>
+    <item fieldId="3" name="Binding" type="DatastoreBindingTargetStruct"/>
+    <item fieldId="4" name="StatusEntry" type="DatastoreStatusEntryStruct"/>
+  </struct>
+
+  <struct name="DatastoreEndpointGroupIDEntryStruct" isFabricScoped="true">
+    <cluster code="0x0752"/>
+    <item fieldId="0" name="NodeID" type="node_id"/>
+    <item fieldId="1" name="EndpointID" type="endpoint_no"/>
+    <item fieldId="2" name="GroupID" type="group_id"/>
+    <item fieldId="3" name="StatusEntry" type="DatastoreStatusEntryStruct"/>
+  </struct>
+
+  <struct name="DatastoreEndpointEntryStruct">
+    <cluster code="0x0752"/>
+    <item fieldId="0" name="EndpointID" type="endpoint_no"/>
+    <item fieldId="1" name="NodeID" type="node_id"/>
```

Comment:
Spec issue: how is someone supposed to make sense of this NodeID without knowing what fabric is involved?  Should this struct be fabric-scoped?

Reply:
https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/11620